src/__tests__/Login.test.js

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { renderWithReduxAndRouter as render } from "../utils/renderWithReduxAndRouter.js";
+import { BrowserRouter as Router } from "react-router-dom";
+import {
+  fireEvent,
+  wait,
+  getAllByText,
+  getAllByAltText,
+} from "@testing-library/react";
+import Login from "../components/Login";
+import axios from "axios";
+import { Provider } from "react-redux";
+
+describe("Login", () => {
+  it("shows login form", () => {
+    render(<Login />);
+  });
+
+  it("renders login fields", () => {
+    const { getByText } = render(<Login />);
+
+    getByText(/email/i);
+    getByText(/Password/i);
+  });
+
+  //mock up
+  let mock;
+  beforeEach(() => {
+    mock = jest.spyOn(axios, "post");
+  });
+  afterEach(() => {
+    mock.mockRestore();
+  });
+
+  const { getByPlaceholderText, getByText, getByTestId } = render(<Login />);
+
+  const fakeForm = (username, password) => {
+    fireEvent.change(getByPlaceholderText("email"), {
+      target: { value: username },
+    });
+    fireEvent.change(getByPlaceholderText(/password/i), {
+      target: { value: password },
+    });
+  };
+
+  it("login with faux data", async () => {
```

Comment:
Great Description - not cryptic at all I can tell what is happening in this test easily